### PR TITLE
Add parameter for eviction strategy

### PR DESCRIPTION
**Pull request by kurnevsky**

### CODE REVIEW

File: src/toxcore/dht/dht_node.rs [hunk 1]
```diff
@@ -234,17 +234,17 @@ mod tests {
...
             let mut bucket = Bucket::new(BUCKET_DEFAULT_SIZE);
-            assert_eq!(true, bucket.try_add(&pk, &n1));
-            assert_eq!(true, bucket.try_add(&pk, &n2));
-            assert_eq!(true, bucket.try_add(&pk, &n3));
-            assert_eq!(true, bucket.try_add(&pk, &n4));
-            assert_eq!(true, bucket.try_add(&pk, &n5));
-            assert_eq!(true, bucket.try_add(&pk, &n6));
-            assert_eq!(true, bucket.try_add(&pk, &n7));
-            assert_eq!(true, bucket.try_add(&pk, &n8));
+            assert_eq!(true, bucket.try_add(&pk, &n1, true));
```
Comment (kpp) [inline src/toxcore/dht/dht_node.rs]: what does `true` mean?

Comment (kurnevsky) [reply src/toxcore/dht/dht_node.rs]: This test should be removed - it tests nothing in this module.

File: src/toxcore/dht/kbucket.rs [hunk 2]
```diff
@@ -477,15 +501,20 @@ impl Kbucket {
     }
 
     /// Create iterator over [`DhtNode`](./struct.DhtNode.html)s in `Kbucket`.
+    /// Nodes that this iterator produces are sorted by distance to a base
+    /// `PublicKey` (in ascending order).
     pub fn iter(&self) -> impl Iterator<Item = &DhtNode> {
         self.buckets.iter()
+            .rev()
```
Comment (kpp) [inline src/toxcore/dht/kbucket.rs]: Why rev?

Comment (kurnevsky) [reply src/toxcore/dht/kbucket.rs]: Because kbucket stores farther nodes at the beginning, but bucket - at the end.

**Pull request merged**